bcipy/signal/model/keyinput.py

```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+
+def compute_keyinput_probs(
+    inquiry: List[str],
+    symbol_set: List[str],
+    user_error_prob: float,
+    proceed: bool,
+):
+    """
+    When the user responds to an inquiry preview, we compute a likelihood term for each symbol
+    based on their response. These likelihood terms will be multiplied with the prior probabilities of
+    each symbol and then normalized, in order to update the symbol probabilities.
+
+    Args:
+        inquiry - the symbols presented in the inquiry preview
+        symbol_set - the full alphabet of symbols
+        proceed - whether or not the user wants to proceed with this inquiry
+        user_error_prob - the probability that the user selects incorrectly during preview
+    """
+    K = len(inquiry)
+    A = len(symbol_set)
+    if not 0 <= user_error_prob <= 1:
+        raise ValueError(f"invalid user error prob: {user_error_prob}")
+
+    if proceed:  # user liked the inquiry; presented letters are upgraded and others are downgraded
+        shown_letter_value = (1 - user_error_prob) / K
+        other_letter_value = user_error_prob / (A - K)
+
+    else:  # user disliked the inquiry; presented are downgraded and others are upgraded
+        shown_letter_value = user_error_prob / K
+        other_letter_value = (1 - user_error_prob) / (A - K)
+
+    return {"KEYINPUT": [shown_letter_value if letter in inquiry else other_letter_value for letter in symbol_set]}
```

Comment:
Let's return the list and ask the caller to construct the dictionary or accept an evidence label that can be used here.

Reply:
Ok - returning list sounds simplest to use imo